lib/puppet/configurer.rb

```diff
@@ -5,6 +5,17 @@
 #require 'puppet/parser/script_compiler'
 require_relative '../puppet/pops/evaluator/deferred_resolver'
 
+# max fact name length in bytes
+MAX_FACT_NAME_LENGTH = 2713
+# max fact value length in bytes
+MAX_FACT_VALUE_LENGTH = 512
```

Comment:
It’s surprising to me that the name limit is bigger than the value limit. I can imagine people do surprising things, such as storing the contents of a file as a fact value. Maybe set the limit to 4KB?